### PR TITLE
CI skip julia and fix doc generation

### PR DESCRIPTION
**Pull request by mathurinm**

### CODE REVIEW

**Comment by tomMoral** [comment]
I don't think this is the problem, the last release of `pyjulia` was on october 27 and the CI breaks start on November 27.
I would bet more on the `julia` package from `conda-forge` that was updated on November 23.

**Comment by mathurinm** [comment]
My bad, I misread the month.

I'm trying with 1.6.3 on the CI now. But it fails locally :( 

`benchopt install . -e --recreate -s julia-pgd`:

```
done
#
# To activate this environment, use
#
#     $ conda activate benchopt_lasso
#
# To deactivate an active environment, use
#
#     $ conda deactivate

# Install
Collecting packages:
- 'Simulated' already available in 'benchopt_lasso'
... done
Installing required packages for:
- Julia-PGD
- finance
- leukemia
- libsvm
...Traceback (most recent call last):
  File "/home/mathurin/anaconda3/bin/benchopt", line 33, in <module>
    sys.exit(load_entry_point('benchopt', 'console_scripts', 'benchopt')())
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/mathurin/workspace/benchopt/benchopt/cli/main.py", line 277, in install
    benchmark.install_all_requirements(
  File "/home/mathurin/workspace/benchopt/benchopt/benchmark.py", line 233, in install_all_requirements
    install_in_conda_env(
  File "/home/mathurin/workspace/benchopt/benchopt/utils/conda_env_cmd.py", line 195, in install_in_conda_env
    _run_shell_in_conda_env(
  File "/home/mathurin/workspace/benchopt/benchopt/utils/shell_cmd.py", line 130, in _run_shell_in_conda_env
    return _run_shell(
  File "/home/mathurin/workspace/benchopt/benchopt/utils/shell_cmd.py", line 68, in _run_shell
    raise RuntimeError(raise_on_error.format(output=output))
RuntimeError: Failed to conda install packages ('julia=1.6.3', 'pip:julia', 'pip:libsvmdata', 'scikit-learn')
Error:Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /home/mathurin/anaconda3/envs/benchopt_lasso

  added / updated specs:
    - julia=1.6.3
    - scikit-learn


The following NEW packages will be INSTALLED:

  arpack             conda-forge/linux-64::arpack-3.7.0-hdefa2d7_2
  bzip2              conda-forge/linux-64::bzip2-1.0.8-h7f98852_4
  c-ares             conda-forge/linux-64::c-ares-1.18.1-h7f98852_0
  curl               conda-forge/linux-64::curl-7.80.0-h494985f_0
  expat              conda-forge/linux-64::expat-2.4.1-h9c3ff4c_0
  gettext            conda-forge/linux-64::gettext-0.19.8.1-h73d1719_1008
  git                conda-forge/linux-64::git-2.34.1-pl5321hf874766_0
  gmp                conda-forge/linux-64::gmp-6.2.1-h58526e2_0
  joblib             conda-forge/noarch::joblib-1.1.0-pyhd8ed1ab_0
  julia              conda-forge/linux-64::julia-1.6.3-h2a39549_1
  krb5               conda-forge/linux-64::krb5-1.19.2-h48eae69_3
  libcurl            conda-forge/linux-64::libcurl-7.80.0-h494985f_0
  libedit            conda-forge/linux-64::libedit-3.1.20191231-he28a2e2_2
  libev              conda-forge/linux-64::libev-4.33-h516909a_1
  libgit2            conda-forge/linux-64::libgit2-1.3.0-haabb1ae_1
  libiconv           conda-forge/linux-64::libiconv-1.16-h516909a_0
  libnghttp2         conda-forge/linux-64::libnghttp2-1.43.0-ha19adfc_1
  libopenblas-ilp64  conda-forge/linux-64::libopenblas-ilp64-0.3.18-pthreads_h0aa5bea_0
  libssh2            conda-forge/linux-64::libssh2-1.10.0-ha35d2d1_2
  libunwind          conda-forge/linux-64::libunwind-1.6.0-h9c3ff4c_1
  libutf8proc        conda-forge/linux-64::libutf8proc-2.6.1-h7f98852_0
  mbedtls            conda-forge/linux-64::mbedtls-3.0.0-h9c3ff4c_0
  metis              conda-forge/linux-64::metis-5.1.0-h58526e2_1006
  mpfr               conda-forge/linux-64::mpfr-4.1.0-h9202a9a_1
  openblas-ilp64     conda-forge/linux-64::openblas-ilp64-0.3.18-pthreads_h3d04fff_0
  openlibm           conda-forge/linux-64::openlibm-0.7.0-h516909a_0
  p7zip              conda-forge/linux-64::p7zip-16.02-he1b5a44_1000
  pcre2              conda-forge/linux-64::pcre2-10.37-h032f7d1_0
  perl               conda-forge/linux-64::perl-5.32.1-1_h7f98852_perl5
  scikit-learn       conda-forge/linux-64::scikit-learn-1.0.1-py38h1561384_2
  scipy              conda-forge/linux-64::scipy-1.7.3-py38h56a6a73_0
  suitesparse        conda-forge/linux-64::suitesparse-5.10.1-h9e50725_1
  tbb                conda-forge/linux-64::tbb-2021.4.0-h4bd325d_1
  threadpoolctl      conda-forge/noarch::threadpoolctl-3.0.0-pyh8a188c0_0


Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... 
done
ERROR conda.core.link:_execute(699): An error occurred while installing package 'conda-forge::julia-1.6.3-h2a39549_1'.
Rolling back transaction: ...working... done

LinkError: post-link script failed for package conda-forge::julia-1.6.3-h2a39549_1
location of failed script: /home/mathurin/anaconda3/envs/benchopt_lasso/bin/.julia-post-link.sh
==> script messages <==
<None>
==> script output <==
stdout: 
stderr: /home/mathurin/anaconda3/envs/benchopt_lasso/bin/.julia-post-link.sh: line 1: 107000 Segmentation fault      (core dumped) "${PREFIX}/bin/julia" -e 'using Pkg; Pkg.instantiate()' >> "${PREFIX}/.messages.txt" 2>&1*
return code: 139

()
```

**Comment by tomMoral** [comment]
Yes and there is this update in the test log when installing `R`...
image

**Comment by tanglef** [comment]
Technically it broke the day `libunwind` updated their version, but this library is also used in a lot of other packages without problem and pyjulia tests can still be run successfully.

**Comment by mathurinm** [comment]
Indeed here they suggest to install `libunwind` 1.5:
https://github.com/conda-forge/julia-feedstock/issues/132

**Comment by mathurinm** [comment]
Why is it still installing `libunwind=1.6` despite being told to use 1.5? https://github.com/benchopt/benchOpt/runs/4394990768?check_suite_focus=true#step:6:321

The install command seems to work fine, but then when testing for the solver installation it says no module named julia, I can't understand why.

File: benchopt/helpers/julia.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ class JuliaSolver(BaseSolver):
 
     # Requirements
     install_cmd = 'conda'
-    requirements = ['julia', 'pip:julia']
+    requirements = ['julia', 'libunwind=1.5' 'pip:julia']
```
Comment (tomMoral) [inline benchopt/helpers/julia.py:54]: A comma is missing
```suggestion
    requirements = ['julia', 'libunwind=1.5', 'pip:julia']
```

**Comment by tomMoral** [comment]
You also need to fix this one in the `R` solver:

image

**Comment by tomMoral** [comment]
Pinning version for conda is tricky, in particular with our `--update-all` that I had to add due to some weird behavior earlier.

**Comment by tomMoral** [comment]
This does not seem to solve the issue either: https://github.com/benchopt/benchOpt/runs/4395558762?check_suite_focus=true
And `libunwind` does not exist on OSX...

**Comment by tanglef** [comment]
It does not solve it because scikit learn updates it back to `1.6.0` ? (We can't really pin it everywhere...)

File: benchopt/helpers/julia.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ class JuliaSolver(BaseSolver):
 
     # Requirements
     install_cmd = 'conda'
-    requirements = ['julia', 'pip:julia']
+    requirements = ['julia', 'libunwind=1.5', 'pip:julia']
```
Comment (tomMoral) [inline benchopt/helpers/julia.py:54]:
```suggestion
    requirements = ['julia', 'pip:julia']
```

**Review by tomMoral — APPROVED** [review]
This is good enough as a hot fix.

**Pull request merged**